pennylane/vqe/vqe.py

```diff
@@ -247,12 +268,13 @@ def _obs_data(self):
         """
         data = set()
 
-        for co, op in zip(*self.terms):
+        coeffs_arr = qml.math.toarray(self.coeffs)
+        for co, op in zip(coeffs_arr, self.ops):
             obs = op.non_identity_obs if isinstance(op, Tensor) else [op]
             tensor = []
             for ob in obs:
                 parameters = tuple(
-                    param.tostring() for param in ob.parameters
+                    str(param) for param in ob.parameters
```

Comment:
Tensors may not have a `tostring` method!

Reply:
ah!